src/main/java/com/griddynamics/jagger/jenkins/jaas/plugin/JaggerTestExecutionBuilder.java

```diff
@@ -0,0 +1,227 @@
+        final PrintStream logger = listener.getLogger();
+
+        logger.println("\n\nJagger JaaS Jenkins Plugin Step 1: Creating TestExecution request...");
+        TestExecutionEntity testExecutionEntity = createTestExecution(logger);
+
+        logger.println("\n\nJagger JaaS Jenkins Plugin Step 2: Sending request to JaaS...");
+        TestExecutionEntity sentExecution = sendTestExecutionToJaas(logger, testExecutionEntity);
+
+        logger.println("\n\nJagger JaaS Jenkins Plugin Step 3: Waiting test to start execution...");
+        waitTestExecutionStarted(logger, sentExecution.getId());
+
+        logger.println("\n\nJagger JaaS Jenkins Plugin Step 4: Waiting test to finish execution...");
+        waitTestExecutionFinished(logger, sentExecution.getId());
+
+        logger.println("\n\nJagger JaaS Jenkins Plugin Step 5: Publishing Test execution results...");
+        logger.println("Test execution report can be found by the link " + jaasEndpoint + "/report?sessionId=valid-session-id");
+    }
+
+    private TestExecutionEntity createTestExecution(PrintStream logger) {
+        TestExecutionEntity testExecutionEntity = new TestExecutionEntity();
+        testExecutionEntity.setEnvId(envId);
+        testExecutionEntity.setLoadScenarioId(loadScenarioId);
+        testExecutionEntity.setExecutionStartTimeoutInSeconds(Long.parseLong(executionStartTimeoutInSeconds));
+        testExecutionEntity.setTestProjectURL(testProjectUrl);
+
+        logger.println(format("JaaS endpoint: %s", jaasEndpoint));
+        logger.println("Test execution properties:");
+        logger.println(format("    Environment ID: %s", envId));
+        if (StringUtils.isNotEmpty(loadScenarioId))
+            logger.println(format("    Load scenario ID: %s", loadScenarioId));
+        if (StringUtils.isNotEmpty(testProjectUrl))
+            logger.println(format("    Test project URL: %s", testProjectUrl));
+        logger.println(format("    Execution start timeout in seconds: %s", executionStartTimeoutInSeconds));
+        return testExecutionEntity;
+    }
+
+    private TestExecutionEntity sendTestExecutionToJaas(PrintStream logger, TestExecutionEntity testExecutionEntity) throws AbortException {
+        try {
+            RequestEntity<TestExecutionEntity> requestEntity = RequestEntity.post(new URI(jaasEndpoint + "/executions")).body(testExecutionEntity);
+            ResponseEntity<TestExecutionEntity> responseEntity = restTemplate.exchange(requestEntity, TestExecutionEntity.class);
+            logger.println("Request successfully sent!\n");
+            logger.println("Response status: " + responseEntity.getStatusCodeValue());
+            logger.println("Response body: \n" + responseEntity.getBody());
+            return responseEntity.getBody();
+        } catch (URISyntaxException e) {
+            logger.println();
+            throw new AbortException("Invalid JaaS endpoint URL: " + e.getMessage());
+        } catch (RestClientException ex) {
```

Comment:
RestServerException can be thrown as well. and I think we need somehow handle it. Retries, for example